genshin/client.py

```diff
@@ -1485,78 +1492,48 @@ async def claim_daily_reward(
 class MultiCookieClient(GenshinClient):
     """A Genshin Client which allows setting multiple cookies"""
 
-    sessions: List[aiohttp.ClientSession]
-
     def __init__(
         self,
         cookie_list: Iterable[Mapping[str, str]] = None,
         *,
         lang: str = "en-us",
         debug: bool = False,
+        session: Optional[aiohttp.ClientSession] = None,
     ) -> None:
-        self.sessions = []
+        self._session = session
 
         if cookie_list:
-            self.set_cookies(cookie_list)
+            self.cookies = list(cookie_list) if cookie_list else []
 
         super().__init__(lang=lang, debug=debug)
 
-    @property
-    def session(self) -> aiohttp.ClientSession:
-        """The currently chosen session"""
-        if not self.sessions:
-            return aiohttp.ClientSession()
-
-        return self.sessions[0]
-
-    @property
-    def cookies(self) -> List[Mapping[str, str]]:
-        """A list of all cookies"""
-        return [{m.key: m.value for m in s.cookie_jar} for s in self.sessions]
-
-    @cookies.setter
-    def cookies(self, cookies: List[Mapping[str, Any]]) -> None:
-        self.set_cookies(cookies)
-
     def set_cookies(
         self,
-        cookie_list: Union[Iterable[Union[Mapping[str, Any], str]], str],
+        cookie_list: Union[Iterable[Mapping[str, Any]], str],
         clear: bool = True,
     ) -> List[Mapping[str, str]]:
         """Set a list of cookies
 
         :param cookie_list: A list of cookies or a json file containing cookies
         :param clear: Whether to clear all of the previous cookies
         """
-        if clear:
-            self.sessions.clear()
-
         if isinstance(cookie_list, str):
             with open(cookie_list) as file:
                 cookie_list = json.load(file)
 
             if not isinstance(cookie_list, list):
                 raise RuntimeError("Json file must contain a list of cookies")
 
-        for cookies in cookie_list:
-            session = aiohttp.ClientSession(cookies=SimpleCookie(cookies))
-            self.sessions.append(session)
+        if clear:
+            self.cookies = list(cookie_list)
+        else:
+            self.cookies.extend(cookie_list)
```

Comment:
A bit late, but I feel I should add the same validation as in GenshinClient here, I'll get on that when I get feedback on how to handle cookies in GenshinClient, as I'll take the same approach here.